lib/codecs/src/encoding/mod.rs

```diff
@@ -182,25 +188,39 @@ impl tokio_util::codec::Encoder<()> for Framer {
 // Unfortunately, copying options of the nested enum variants is necessary
 // since `serde` doesn't allow `flatten`ing these:
 // https://github.com/serde-rs/serde/issues/1402.
-#[derive(Debug, Clone, Deserialize, Serialize)]
+#[configurable_component]
+#[derive(Clone, Debug)]
 #[serde(tag = "codec", rename_all = "snake_case")]
 pub enum SerializerConfig {
-    /// Configures the `AvroSerializer`.
+    /// Apache Avro serialization.
     Avro {
-        /// Options for the avro serializer.
+        /// Apache Avro serializer options.
         avro: AvroSerializerOptions,
     },
-    /// Configures the `JsonSerializer`.
+
+    /// JSON serialization.
     Json,
-    /// Configures the `LogfmtSerializer`.
+
+    /// Logfmt serialization.
     Logfmt,
-    /// Configures the `NativeSerializer`.
+
+    /// Native Vector serialization based on Protocol Buffers.
     Native,
-    /// Configures the `NativeJsonSerializer`.
+
+    /// Native Vector serialization based on JSON.
     NativeJson,
-    /// Configures the `RawMessageSerializer`.
+
+    /// No serialization.
+    ///
+    /// This emits the raw bytes for a given input.
```

Comment:
That would be cool and has been the intended meaning. However, we couldn't do it until @fuchsnj's work on the log namespacing is finished to allow bytes as a top-level value.

So currently it also just reads the `message` field...

https://github.com/vectordotdev/vector/blob/fd0b6c19021b5b96ea355cb15ebfd9b988b9f965/lib/codecs/src/encoding/format/raw_message.rs#L52

Reply:
🙈 I'll update the comment.

Reply:
Fixed in 2d2b23f55.